### PR TITLE
fixed unchecked subtractions Instant - Duration

### PR DESCRIPTION
**Pull request by pompon0**
Subtraction Instant - Duration is not saturating, it panics instead. Perhaps we should rather make instant-Duration return Option\<Instant\> instead?
Also migrated config fields from std::time to near_network_primitives::time

### CODE REVIEW

**Review by matklad — APPROVED** [review]
Changes to use our `time` LGTM! 

Changes to `.checked_sub` feel suspicious -- I wish we just didn't have to think about this case and I'd argue that's the bug in mac version of stdlib.

Really torn on whether we should work-around a mac "bug" here. 

OTOH, I think either way it's going to be only a small mistake, so r+. And let me create a rust-lang issue for this...

File: chain/network/src/peer_manager/tests.rs [hunk 1]
```diff
@@ -326,7 +326,7 @@ async fn accounts_data_rate_limiting() {
     let mut pms = vec![];
     for _ in 0..n * m {
         let mut cfg = chain.make_config(rng);
-        cfg.accounts_data_broadcast_rate_limit = demux::RateLimit { qps: 0.5, burst: 1 };
+        cfg.accounts_data_broadcast_rate_limit = demux::RateLimit { qps: 1., burst: 1 };
```
Comment (mm-near) [inline chain/network/src/peer_manager/tests.rs:329]: this doesn't seem to be related to thePR description

**Review by mm-near — APPROVED** [review]

File: chain/network/src/testonly/actix.rs [hunk 2]
```diff
@@ -12,6 +12,9 @@ impl Thread {
 
 impl Drop for Thread {
     fn drop(&mut self) {
+        if std::thread::panicking() {
+            return;
+        }
         self.0.take().unwrap().join().unwrap().unwrap();
```
Comment (matklad) [inline chain/network/src/testonly/actix.rs:18]:
```suggestion
        let h = self.0.take().unwrap().join();
        if !std::thread::panicking() {
            h.unwrap().unwrap();
        }
```

Joining even with panic would be marginally better I think

**Review by matklad — APPROVED** [review]

File: chain/network/src/peer_manager/tests.rs (hunk 1 above)
Comment (pompon0) [reply chain/network/src/peer_manager/tests.rs]: yeah, moved to a separate PR

File: chain/network/src/testonly/actix.rs (hunk 2 above)
Comment (pompon0) [reply chain/network/src/testonly/actix.rs:18]: moved to a separate PR
